### ISSUE TITLE
Replicating the 130M Perplexity result

### ISSUE THREAD

**Issue by shikhar-srivastava** [lmsdss/LayerNorm-Scaling#5]
Dear @lmsdss and @Shiweiliuiiiiiii,

Congratulations on this great work.

I’m hoping to replicate the **25.76** perplexity result for the **LLAMA-130M** model from the paper. However, the `run_130m.sh` script uses `max_training_steps = 20,000`, which results in approximately **2.08 B tokens**, not **2.2 B**.

Since the codebase’s `max_train_tokens` logic is:

```bash
args.num_training_steps = args.max_train_tokens // args.total_batch_size
```
It doesn't help setting the max tokens correctly. 

Would help immensely if you can provide the exact replication configuration.

Thank you, and congratulations once again.

**Comment by Shiweiliuiiiiiii** [comment]
Hi, I believe this is exactly the configuration we used in our paper. Our model was also trained with 2.08B tokens.

**Comment by shikhar-srivastava** [comment]
Hi, 
Per the figure in the repository (below), its **2.2B** tokens for the 130M parameter model. Also, running the provided replication for 130M: `bash run_130m.sh LNS 3` gives a final validation loss of **3.26633** or **26.215** perplexity.

Note that the figure below shows perplexity of **25.76 for 130M**. 
Hence my query. 

<img width="1090" height="370" alt="Image" src="https://github.com/user-attachments/assets/6be4331d-5051-4734-9db6-fddb49d67165" />
